### ISSUE TITLE
Taproot on-chain addresses in Core Lightning?

### ISSUE THREAD

**Issue by schmitmd** [ZeusLN/zeus#1698]
### Describe the feature

As of version 23.08 released a little over a month ago, Core Lightning now supports native taproot on-chain address. Are there plans to add support for that to Zeus?

**Comment by kaloudis** [comment]
We'd like to do this but we first want to implement the new CLN-rest backend and/or Commando+RPC

**Comment by schmitmd** [comment]
Superseded by #1637 and https://github.com/ZeusLN/zeus/blob/master/docs/Bounties.md#core-lightning-commandolnmessage-connection

**Comment by schmitmd** [comment]
Just to check in @kaloudis 
Now that native CLNrest is supported, will this feature still need to wait until the whole Commando+RPC backend is done or can it be tacked in..?

**Comment by schmitmd** [comment]
@kaloudis ? Anyone?

**Comment by kaloudis** [comment]
https://docs.corelightning.org/reference/newaddr

**Comment by schmitmd** [comment]
Yeah, of course.  But I can't "receive" to it from within Zeus.  The option is available for LND (embedded) but not for c-lightning.

**Comment by schmitmd** [comment]
I'm sure you're very busy with v0.10, so I really don't want to pester, but from the comments so far, I can't tell if this is a "not yet" or a "not ever".  Generating a taproot address from lightning-cli is easy enough (as linked by @kaloudis) but interacting with it from Zeus is still a ? for me.  Am I just not seeing something obvious or..?

**Comment by kaloudis** [comment]
not yet

**Comment by kaloudis** [comment]
> Hey [@kaloudis](https://github.com/kaloudis) when Taproot address support is added for CLN, would it make sense to let users choose between generating a native SegWit or Taproot address?

Yes. Please examine how we have done so with LND nodes.

**Comment by a-khushal** [comment]
@kaloudis please review #3005
